astropy/coordinates/tests/test_regression.py

```diff
@@ -751,3 +751,70 @@ def test_regression_10291():
     assert_quantity_allclose(
         venus.separation(sun), 554.427 * u.arcsecond, atol=0.001 * u.arcsecond
     )
+
+
+@pytest.mark.parametrize("coord_cls", [ICRS, SkyCoord])
+class TestRegression16998:
+    @pytest.mark.parametrize(
+        "extra_kwargs", [{}, dict(representation_type="unitspherical")]
+    )
+    def test_unitspherical_ra_dec(self, coord_cls, extra_kwargs):
+        """
+        Ensure that frames/coordinates in UnitSpherical are represented
+        correctly when writing to files.
+        """
+        coord = coord_cls([0, 10] * u.deg, [20, 30] * u.deg, **extra_kwargs)
+        # Direct regression test that we get dec in a file.
+        out = io.StringIO()
+        Table({"coord": coord}).write(out, format="ascii.ecsv")
+        assert "dec" in out.getvalue()
+        # Check of original problem, that "dec" was not present.
+        as_dict = coord.info._represent_as_dict()
+        assert "ra" in as_dict
+        assert "dec" in as_dict
+        assert "distance" not in as_dict
```

Comment:
I included the test of writing out the table since that was the one in the original issue... And this is fine in the sense that really `_represent_as_dict()` exists only for serializing a mixin column to a set of regular columns. But obviously, it is a bit silly to keep on creating tables when we know what the problem is, so I settled for making an exception to the testing of private methods, putting it in `test_regression` on purpose since I couldn't think of a nice, proper way to test...

Maybe two options: add a comment here along the lines of what I wrote above, or add a test column to `astropy/io/tests/mixin_columns.py`. Happy to go either way. 

Anyway, I basically agree with you that one should not test private